src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
-export interface Figure {
+type Shape = 'rectangle' | 'circle' | 'triangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  color: Color;
+
+  shape: Shape = 'triangle';
+
+  constructor(color: Color, sideA: number, sideB: number, sideC: number) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('Some length is less than zero');
```

Comment:
The error message is not accurate. It should be 'Some length is less than or equal to zero'.